Objects/unicodeobject.c

```diff
@@ -14794,8 +14794,10 @@ unicodeiter_reduce(unicodeiterobject *it, PyObject *Py_UNUSED(ignored))
         return Py_BuildValue("N(O)n", iter, it->it_seq, it->it_index);
     } else {
         PyObject *u = unicode_new_empty();
-        if (u == NULL)
+        if (u == NULL) {
+            Py_DECREF(iter);
```

Comment:
Shouldn't this be `Py_XDECREF`? There is no null check above for iter.

Reply:
Good point. But shouldn’t it be a null check and return above, if iter is null I think it would crash in the BuildValue before we even get here.

Reply:
`Py_BuildValue` deals with NULL internally; if one of the args is NULL it itself returns NULL, propagating the exception. So changing to XDECREF here should be enough.

Reply:
#102283